src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/AutoInsert/AutoClosingTagOnAutoInsertProvider.cs

```diff
@@ -249,6 +248,13 @@ private static bool TryEnsureOwner_WorkaroundCompilerQuirks(int afterCloseAngleI
                 //
                 // The owner will be the </div>. Note this does not happen outside of C# blocks.
 
+                var closeAngleSourceChange = new SourceChange(markup.StartTag.EndPosition + 1, length: 0, newText: string.Empty);
```

Comment:
With this new change what's the new owner end up being?

Reply:
The idea of the algorithm here is the following: instead of getting the owner of `source location - 1` I serach for the opening tag of the "parent" and get the actual owner by stepping 1 char next. In the given example `<div><a target="_blank"|</div>` `div` element is recognized as an owner. In the old logic you step 1 char left and get quote -> locate an attribute -> fail to complete the tag. In my algo you first get the opening `<div>` and locate 1 char right -> get `<` of the actual tag -> close it. When writing this I realised that this example would probably break my logic as well:
`<div> <a target="_blank"</div>` - there is a space between the `<div>` and opening of `a`. No sure if it is possible to get trivia length here to add it to the search offset as well.

Reply:
Ah I see what you were going for. I think this also breaks in: `<p><br /><em>|</p>` because it'll incorrectly see the `<br />`